frontend/src/diagram/DiagramWebSocketContainer.tsx

```diff
@@ -415,12 +417,15 @@ export const DiagramWebSocketContainer = ({ representationId, selection, setSele
   ]);
 
   useEffect(() => {
-    if (!toolSectionData.loading && viewState === READY__STATE) {
-      if (toolSectionData?.data?.data?.viewer?.toolSections) {
-        dispatch({ type: SET_TOOL_SECTIONS__ACTION, toolSections: toolSectionData.data.data.viewer.toolSections });
+    if (!toolSectionLoading && viewState === READY__STATE) {
+      if (toolSectionData?.viewer?.toolSections) {
+        dispatch({
+          type: SET_TOOL_SECTIONS__ACTION,
+          toolSections: JSON.parse(JSON.stringify(toolSectionData.viewer.toolSections)),
```

Comment:
The reducer should be modified to use what it needs from the server response instead of just modifying it inline.

JSON.parse + JSON.stringify is overkill to not use some basic rest and spread.

Reply:
Fixed by https://github.com/eclipse-sirius/sirius-components/pull/128/commits/44ce7aede446bb4bced5cd8e72c5bb2d66911ef3